mailpoet/assets/js/src/email-editor/engine/components/header/more-menu.tsx

```diff
@@ -72,9 +79,20 @@ export function MoreMenu(): JSX.Element {
               />
             </MenuGroup>
             <MenuGroup>
-              <MenuItem onClick={() => setShowTrashModal(true)} isDestructive>
-                {__('Move to trash', 'mailpoet')}
-              </MenuItem>
+              {status === 'trash' ? (
+                <MenuItem
+                  onClick={async () => {
+                    await setStatus('draft');
```

Comment:
I think changing the post status is not enough because the newsletter entity stays marked as deleted. We also have to set the property `deletedAt` as null. This means that after restoring from the menu, the newsletter is still in the trash on the listing.

Reply:
Should we remove it then? We don't have an endpoint in place to sync and update both entities at the same time.

Reply:
When we extend the `mailpoet_data` in [EmailApiController](https://github.com/mailpoet/mailpoet/blob/55aae2c4a12dc5303249a6b9ee05bdaa56b5d331/mailpoet/lib/EmailEditor/Integrations/MailPoet/EmailApiController.php) it should work. After that, you can add updating mailpoet email property deleted_at: `await updateEmailMailPoetProperty('deleted_at', null);`. What do you think?

Reply:
@lysyjan Done in https://github.com/mailpoet/mailpoet/pull/5532/commits/11f4f8f8c7ca0f0b87d24ca5e9fd7af2eada3e44